satpy/readers/yaml_reader.py

```diff
@@ -498,6 +520,14 @@ def create_filehandlers(self, filenames):
                     filehandlers,
                     key=lambda fhd: (fhd.start_time, fhd.filename))
 
+        # Abort if there are insufficient requirements to read any data
+        # (e.g. no file handlers at all or only requirements).
+        only_requirements = all([self._is_requirement(filetype)
+                                 for filetype in self.file_handlers])
+        if not self.file_handlers or only_requirements:
+            raise RuntimeError('Insufficient requirements (e.g. Prolog, '
+                               'Epilog) to read any data')
```

Comment:
This assumes that required files don't have any other useful data in them. While that might be the case for the Prolog/Epilog-based readers that may not be true for all future readers. In the Scene we *could* do a `self.available_dataset_names()` check and if filenames were provided and no datasets are available then raise an exception. @mraspaud thoughts?

If this *has* to be done here then I would prefer that a reverse requirements lookup dictionary be created when the Reader is created so that the `_is_requirement` function is just a dictionary `filetype in self.filetype_required_by` or something. This loop is `O(N^2)` (if I'm correct) so any way to reduce how many times the file types are iterated over would be great.

Lastly, I'd be ok with this type of error being ignored here. Technically a reader could have files added to it over time and not all in one call (I think) so this check may be premature. The `not self.file_handlers` is handled by the loader already so it wouldn't need to be here too if the requirements check was removed. I'm open to this if it can be done quickly and not slow down other readers (that don't have required file types), just convince me :wink: .

Reply:
Oh, many good points! I have to think about that. Not sure my understanding of the whole process is good enough to come up with a general solution.

Reply:
Agree, the check in the reader is premature. Your idea to check ``self.available_dataset_names()`` (or ``_ids()`` which would be a little bit faster) in ``Scene.__init__`` is very good. It is much simpler and leads to the same behaviour.